test/unit/core/simd/simd_algorithm_test.cpp

```diff
@@ -38,3 +41,236 @@ TEST(simd_algorithm, iota)
     constexpr simd_type result = iota<simd_type>(0);
     SIMD_EQ(result, expect);
 }
+
+TEST(simd_algorithm, transpose)
+{
+    using simd_t = simd_type_t<uint8_t>;
+
+    if constexpr (simd_traits<simd_t>::length > 1)
+    {
+        std::array<simd_t, simd_traits<simd_t>::length> matrix;
+
+        for (size_t i = 0; i < matrix.size(); ++i)
+            matrix[i] = simd::iota<simd_t>(0);
+
+        simd::transpose(matrix);
+
+        for (size_t i = 0; i < matrix.size(); ++i)
+            SIMD_EQ(matrix[i], simd::fill<simd_t>(i));
+    }
+}
+
+//-----------------------------------------------------------------------------
+// Algorithm load
+//-----------------------------------------------------------------------------
+
+template <typename simd_t>
+struct simd_algorithm_load : ::testing::Test
+{
+
+    void SetUp()
+    {
+        memory.resize(100);
+        std::iota(memory.begin(), memory.end(), 0);
+    }
+
+    std::vector<typename simd_traits<simd_t>::scalar_type> memory;
+};
+
+using simd_load_types = ::testing::Types<simd_type_t<int8_t>,
+                                         simd_type_t<uint8_t>,
+                                         simd_type_t<int16_t>,
+                                         simd_type_t<uint16_t>,
+                                         simd_type_t<int32_t>,
+                                         simd_type_t<uint32_t>,
+                                         simd_type_t<int64_t>,
+                                         simd_type_t<uint64_t>>;
+
+TYPED_TEST_CASE(simd_algorithm_load, simd_load_types);
+
+TYPED_TEST(simd_algorithm_load, load)
+{
+    SIMD_EQ(simd::load<TypeParam>(this->memory.data()), simd::iota<TypeParam>(0));
+    SIMD_EQ(simd::load<TypeParam>(this->memory.data() + 10), simd::iota<TypeParam>(10));
+}
+
+//-----------------------------------------------------------------------------
+// Algorithm extract
+//-----------------------------------------------------------------------------
+
+template <typename simd_t>
+struct simd_algorithm_extract : ::testing::Test
+{};
+
+using simd_extract_types = ::testing::Types<simd_type_t<uint8_t>,
+                                            simd_type_t<uint16_t>,
+                                            simd_type_t<int32_t>,
+                                            simd_type_t<int64_t>>;
+TYPED_TEST_CASE(simd_algorithm_extract, simd_extract_types);
+
+TYPED_TEST(simd_algorithm_extract, extract_halve)
+{
+    TypeParam vec = simd::iota<TypeParam>(0);
+
+    TypeParam fst = detail::extract_halve<0>(vec);
+    TypeParam sec = detail::extract_halve<1>(vec);
+
+    if constexpr (simd_traits<TypeParam>::length == 1)
+    {
+        SIMD_EQ(fst, vec);
+        SIMD_EQ(sec, vec);
+    }
+    else
+    {
+        for (size_t idx = 0; idx < simd_traits<TypeParam>::length / 2; ++idx)
```

Comment:
Really? This feels like a little overeager. I am happy to make this changes in productive code, but for tests I don't see the point of reducing some if clauses. Also I think it is less readable because you have to think twice about why this + 1. I would at least nee to add a comment to explain the +1.

Reply:
i changed it